File: appkey-service/src/main/java/org/appkey/service/AppKeyService.java
package org.appkey.service;

import org.appkey.dao.AppkeyRepository;
import org.appkey.data.UserAppkey;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class AppKeyService {


    @Autowired
    private AppkeyRepository repository;

    public UserAppkey findByPk(String appkey) {
        UserAppkey vo = repository.findById(appkey).get();
        return vo;
    }

    public UserAppkey save(UserAppkey vo) {
        return  repository.save(vo);
    }


    public void delete(UserAppkey vo) {
        repository.delete(vo);
    }
}

File: appkey-service/src/main/kotlin/org/appkey/service/AppKeyService.kt
package org.appkey.service

import org.springframework.beans.factory.annotation.Autowired
import org.appkey.dao.AppkeyRepository
import org.appkey.data.UserAppkey
import org.springframework.stereotype.Service

@Service
class AppKeyService {

    @Autowired
    private lateinit var repository: AppkeyRepository

    fun findByPk(appkey: String): UserAppkey {
        return repository.findById(appkey).get()
    }

    fun save(vo: UserAppkey): UserAppkey? {
        return repository.save(vo)
    }

    fun delete(vo: UserAppkey) {
        repository.delete(vo)
    }
}